src/sentry/tasks/on_demand_metrics.py

```diff
@@ -423,6 +423,9 @@ def check_field_cardinality(
     is_task: bool = False,
     widget_query: DashboardWidgetQuery | None = None,
 ) -> dict[str, str]:
+    if not features.has("organizations:on-demand-metrics-extraction-widgets", organization):
```

Comment:
@k-fish would you be able to advise if this is okay for us to do? I suppose when we migrate fully to EAP then we can get rid of this, but in an effort to make some easy progress for other users, I figured we could just feature flag this check

It seems like we wanted to evaluate it even without the feature flag so when we rolled out, people's queries were good to go

Reply:
We can get rid of the flag if we confirm if any of the orgs are not still on am1. 

In theory yes, we were supposed to check so that we could roll it out for every org, but am3 came along. We're okay skipping it for extraction widgets. 

Can you copy the test with the flag to assert the old behaviour? otherwise it's fine I think